PythonVisualizations/Stack.py

```diff
@@ -0,0 +1,322 @@
+import time
+from tkinter import *
+
+try:
+    from drawable import *
+    from VisualizationApp import *
+except ModuleNotFoundError:
+    from .drawable import *
+    from .VisualizationApp import *
+
+class Stack(VisualizationApp):
+    nextColor = 0
+    MAX_ARG_WIDTH = 8
+    MAX_SIZE = 8
+    CELL_WIDTH = 20 * MAX_ARG_WIDTH
+    CELL_HEIGHT = 40
+    CELL_BORDER = 2
+    CELL_BORDER_COLOR = 'black'
+    STACK_X0 = 300
+    STACK_Y0 = 400
+
+    def __init__(self, size=MAX_SIZE, maxArgWidth=MAX_ARG_WIDTH, title="Stack", **kwargs):
+        super().__init__(title=title, maxArgWidth=maxArgWidth, **kwargs)
+        self.size = size
+        self.title = title
+        self.list = []
+        self.buttons = self.makeButtons()
+        self.maxArgWidth = maxArgWidth
+        self.display()
+
+    def __str__(self):
+        return str(self.list)
+
+    # Create an index arrow to point at an indexed cell with an optional name label
+    def createIndex(self, index, name=None):
+        cell_coords = self.cellCoords(index)
+        cell_center = self.cellCenter(index)
+        x0 = self.STACK_X0 - self.CELL_WIDTH * 4 // 5
+        x1 = self.STACK_X0 - self.CELL_WIDTH * 3 // 10
+        y0 = y1 = cell_coords[1] + self.CELL_HEIGHT // 2
+
+        return self.drawArrow(
+            x0, y0, x1, y1, self.VARIABLE_COLOR, self.VARIABLE_FONT, name=("item", index))
+
+    # draw the actual arrow
+    def drawArrow(
+            self, x0, y0, x1, y1, color, font, name=None):  #
+        arrow = self.canvas.create_line(
+            x0, y0, x1, y1, arrow="last", fill=color)
+        if name:
+            label = self.canvas.create_text(
+                x0 - self.CELL_WIDTH / 2, y0 + self.CELL_HEIGHT / 5, text=name, anchor=SW,
+                font=font, fill=color)
+
+        return (arrow, label) if name else (arrow,)
+
+    # STACK FUNCTIONALITY
+
+    def push(self, val):
+
+        self.startAnimations()
+        callEnviron = self.createCallEnvironment()
+        # draw an index pointing to the last cell
+        indexDisplay = self.createIndex(len(self.list))
+        callEnviron |= set(indexDisplay)
+
+        cellCoords = self.cellCoords(len(self.list))
+        cellCenter = self.cellCenter(len(self.list))
+
+        # create new cell and cell value display objects
+        cellCoords = add_vector(cellCoords, (0, 0, 0, 2))
+        toPositions = (cellCoords, cellCenter)
```

Comment:
The rationale for adjusting the `cellCoords` lower right corner by 2 is unclear.  Was this to try to avoid gaps between the cells?  I think this has something to do with the cell border so at least it should reference `self.CELL_BORDER` instead of 2.
You may not need an adjustment here if you change the definition of `cellCoords` as mentioned below.